src/asus-defines.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+// global asus defines
+
+#define ASUS_QUIRK_DOUBLE_DPI 1 << 0
+#define ASUS_QUIRK_STRIX_PROFILE 1 << 1
+
+#define ASUS_PACKET_SIZE 64
+#define ASUS_BUTTON_ACTION_TYPE_KEY 0  // keyboard key
+#define ASUS_BUTTON_ACTION_TYPE_BUTTON 1  // mouse button
+#define ASUS_BUTTON_CODE_DISABLED 0xff  // disabled mouse button
+#define ASUS_STATUS_ERROR 0xffaa  // invalid state/request, disconnected or sleeping
+
+// maximum number of buttons across all ASUS devices
+#define ASUS_MAX_NUM_BUTTON 16
+
+// maximum number of DPI presets across all ASUS devices
+// for 4 DPI devices: 0 - red, 1 - purple, 2 - blue (default), 3 - green
+// for 2 DPI devices: 0 - main (default), 1 - alternative
+#define ASUS_MAX_NUM_DPI 4
+
+// maximum number of LEDs across all ASUS devices
+#define ASUS_MAX_NUM_LED 3
```

Comment:
afaict both actual source files include asus-defines and asus-common.h. I dont' think we have any future uses for just these defines here in libratbag so let's just have one header instead.

Reply:
If I move everything into a single .h file the compiler would complain about unused stuff while trying to compile the libratbag-data.

Reply:
whats the error message you're getting?

Reply:
It's not an error, but annoying warning:
```
[2/16] Compiling C object libratbag.a.p/src_libratbag-data.c.o
In file included from ../src/libratbag-data.c:33:
../src/asus-common.h:163:21: warning: 'ASUS_POLLING_RATES' defined but not used [-Wunused-variable]
  163 | static unsigned int ASUS_POLLING_RATES[] = { 125, 250, 500, 1000 };
      |                     ^~~~~~~~~~~~~~~~~~
../src/asus-common.h:135:22: warning: 'ASUS_KEY_MAPPING' defined but not used [-Wunused-variable]
  135 | static unsigned char ASUS_KEY_MAPPING[] = {
      |                      ^~~~~~~~~~~~~~~~
```

Reply:
those arrays need to go into `src/driver-asus.c` - that's where they're being used. These two are very libratbag-driver specific since they just convert (or provide) some format libratbag requires that the device itself doesn't have in that format.

So moving them should get rid of the warning too then.